auslib/blobs/gmp.py

```diff
@@ -61,6 +61,15 @@ def getInnerXML(self, updateQuery, update_type, whitelistedDomains, specialForce
 
         return vendorXML
 
+    def validate(self, *args, **kwargs):
+        Blob.validate(self, *args, **kwargs)
+        hashes = dict(SHA1=40, SHA256=64, SHA512=128)
+        required_length = hashes[self["hashFunction"]]
```

Comment:
@mozbhearsum can you confirm if `self["hashFunction"]` will always be upper case, if not I should use `.upper()` for it to get NoSuchKey type of error.